pkg/iniparser_test.go

```diff
@@ -0,0 +1,447 @@
+package iniparser
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+)
+
+const validIni = `[Simple Values]
+you can also use=to delimit keys from values
+key=value
+paces in keys=allowed
+
+[You can use comments]
+# like this
+; or this
+# By default only in an empty line.
+# That being said, this can be customized.`
+
+func TestLoadFromString(t *testing.T) {
+	tests := []struct {
+		name     string
+		data     string
+		expected map[string]section
+		err      bool
+	}{
+		{
+			name: "test on valid INI",
+			data: validIni,
+			expected: map[string]section{
+				"Simple Values": {
+					"you can also use": "to delimit keys from values",
+					"key":              "value",
+					"paces in keys":    "allowed",
+				},
+				"You can use comments": {},
+			},
+			err: false,
+		},
+		{
+			name: "empty section name",
```

Comment:
Add a test for an invalid section as follows ```[ccc```

Reply:
added more tests